fix: rewrite checkbox to use button instead of input

goetzrobin opened the pull request:
I switched the underlying implementation from a hidden input to a button. similar to the switch primitive. this fixes some a11y issues that occured with the previous implementation.

closes #644

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our
      guidelines: https://github.com/goetzrobin/spartan/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Which package are you modifying?

- [ ] accordion
- [ ] alert
- [ ] alert-dialog
- [ ] aspect-ratio
- [ ] avatar
- [ ] badge
- [ ] breadcrumb
- [ ] button
- [ ] calendar
- [ ] card
- [ ] carousel
- [x] checkbox
- [ ] collapsible
- [ ] combobox
- [ ] command
- [ ] context-menu
- [ ] data-table
- [ ] date-picker
- [ ] dialog
- [ ] dropdown-menu
- [ ] form-field
- [ ] hover-card
- [ ] icon
- [ ] input
- [ ] input-otp
- [ ] label
- [ ] menubar
- [ ] navigation-menu
- [ ] pagination
- [ ] popover
- [ ] progress
- [ ] radio-group
- [ ] scroll-area
- [ ] select
- [ ] separator
- [ ] sheet
- [ ] skeleton
- [ ] slider
- [ ] sonner
- [ ] spinner
- [ ] switch
- [ ] table
- [ ] tabs
- [ ] textarea
- [ ] toast
- [ ] toggle
- [ ] toggle-group
- [ ] tooltip
- [ ] typography

## What is the current behavior?

<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Closes #644, #324

## What is the new behavior?

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->

## Other information

Commit:
fix: rewrite checkbox to use button instead of input

I switched the underlying implementation from a hidden input
to a button. similar to the switch primitive. this fixes some
a11y issues that occured with the previous implementation.

closes #644

ashley-hunter commented on libs/brain/checkbox/src/lib/brn-checkbox.component.ts:
```diff
@@ -30,72 +33,62 @@ export const BRN_CHECKBOX_VALUE_ACCESSOR = {
 	},
 	providers: [BRN_CHECKBOX_VALUE_ACCESSOR],
 	changeDetection: ChangeDetectionStrategy.OnPush,
 	encapsulation: ViewEncapsulation.None,
```
I think this can be removed here as this component has no styles.

goetzrobin replied:
I will check it out. I thought this was necessary to "redirect: the aria-label and aria-describedby inputs to the child, but I could be wrong

ashley-hunter replied:
Hmm, GitHub seems to have highlighted different lines than I selected, I was only referring to the view encapsulation line, the rest are perfect 😊

goetzrobin replied:
That makes perfect sense! I'll remove it!

ashley-hunter reviewed (approved):
Thanks! A great update, really simplifies things! I left a few comments on things - none of which you actually changed, but just "tidy" up things in this area - but they can be done in a separate PR if you'd rather.

I noticed this example is still present but isn't relevant any more as we removed this input a while back.

<img width="503" alt="image" src="https://github.com/user-attachments/assets/c0151edf-83e1-4cca-8fdd-d38a5ebe4aaf" />

Commit:
fix: address PR comments and improve docs

removed own icon example
removed doubled event.preventDefaultCall
removed ViewEncapsulation.None
improve comments for auto generated docs
added onTouched calls and touched output during toggle

ashley-hunter reviewed (approved):
Looks great, thank you for the updates! Big improvement!
